sklearn/feature_selection/rfe.py

```diff
@@ -367,47 +385,40 @@ def fit(self, X, y):
                           "value is set via the estimator initialisation or "
                           "set_params method.", DeprecationWarning)
         # Initialization
-        rfe = RFE(estimator=self.estimator, n_features_to_select=1,
-                  step=self.step, estimator_params=self.estimator_params,
-                  verbose=self.verbose - 1)
-
         cv = check_cv(self.cv, X, y, is_classifier(self.estimator))
         scorer = check_scoring(self.estimator, scoring=self.scoring)
-        scores = np.zeros(X.shape[1])
-        n_features_to_select_by_rank = np.zeros(X.shape[1])
+        n_features = X.shape[1]
+        n_features_to_select = 1
+
+        # Determine the number of subsets of features
+        scores = []
 
         # Cross-validation
         for n, (train, test) in enumerate(cv):
             X_train, y_train = _safe_split(self.estimator, X, y, train)
             X_test, y_test = _safe_split(self.estimator, X, y, test, train)
 
+            rfe = RFE(estimator=self.estimator, n_features_to_select=1,
+                      step=self.step, estimator_params=self.estimator_params,
+                      verbose=self.verbose - 1)
+
             # Compute a full ranking of the features
             # ranking_ contains the same set of values for all CV folds,
             # but perhaps reordered
-            ranking_ = rfe.fit(X_train, y_train).ranking_
-            # Score each subset of features
-            for k in range(0, np.max(ranking_)):
-                indices = np.where(ranking_ <= k + 1)[0]
-                estimator = clone(self.estimator)
-                estimator.fit(X_train[:, indices], y_train)
-                score = _score(estimator, X_test[:, indices], y_test, scorer)
-
-                if self.verbose > 0:
-                    print("Finished fold with %d / %d feature ranks, score=%f"
-                          % (k + 1, np.max(ranking_), score))
-                scores[k] += score
-                # n_features_to_select_by_rank[k] is being overwritten
-                # multiple times, but by the same value
-                n_features_to_select_by_rank[k] = indices.size
-
-        # Select the best upper bound for feature rank. It's OK to use the
-        # last ranking_, as np.max(ranking_) is the same over all CV folds.
-        scores = scores[:np.max(ranking_)]
-        k = np.argmax(scores)
-
+            rfe._fit(X_train, y_train, lambda estimator, features:
+                     _score(estimator, X_test[:, features], y_test, scorer))
+            scores.append(np.array(rfe.scores_[::-1]).reshape(1, -1))
+        scores = np.sum(np.concatenate(scores, 0), 0)
+        # The index in scores when 'n_features' features is selected
+        n_feature_index = np.ceil((n_features - n_features_to_select) /
```

Comment:
It might possibly be easier and less prone to bugs to store this? I remember the formula for this was quite tricky.